apps/aecore/lib/aecore/peers/peer_connection.ex

```diff
@@ -529,27 +530,148 @@ defmodule Aecore.Peers.PeerConnection do
     ]
   end
 
-  def rlp_encode(@ping, %{
-        share: share,
-        genesis_hash: genesis_hash,
-        best_hash: best_hash,
-        difficulty: difficulty,
-        peers: peers,
-        port: port
-      }) do
+  # RLP for peer messages
+
+  defp rlp_encode(@p2p_response, %{result: result, type: type, object: object, reason: reason}) do
+    serialized_result = bool_bin(result)
+
+    reason =
+      case reason do
```

Comment:
If `reason` is a String variable, this could be interchanged with the following: 
```elixir
reason = reason || <<>>
```
This is going to work only if `reason`cannot be `false`.